### DISCUSSION TITLE
Teensy 4.1 & ILI9341 TFT (no touch & no CS pin)

### DISCUSSION THREAD

**Discussion by asouth** [ImpulseAdventure/GUIslice#377] [General]
Hi,

I've verified my TFT Screen works with Teensy 4.1 using the script at BOTTOM of this post which does not involve GUISlice in anyway. In the IDE if you set port to "Audio" it will allow you to use Teensy 4.1 as a Digital Audio Interface for playback and recording where the right channel has a low pass filter, while display some text on the screen.

Note that my ILI9341 TFT does not have a CS pin hence why it is set to -1, as per the library recommendation.

Without changing the wiring from the above working project, my issue is that after setting up the simplest GUISlice Builder project: one text box and Target Platform set to "Teensy" see image attached, the project does not copile for the Teensy board ![image](https://user-images.githubusercontent.com/10164856/117349314-2a72c900-aea3-11eb-9d33-a7f0dd768ae5.png), 

In GUISlice_config.h I also uncommented only 
`#include "../configs/teensy-adagfx-ili9341_t3-notouch-audio.h` 
and set pinout as

```c++
  // -----------------------------------------------------------------------------
  // SECTION 2: Pinout
  // -----------------------------------------------------------------------------

  // For shields, the following pinouts are typically hardcoded
  #define ADAGFX_PIN_CS       -1     // Display chip select
  #define ADAGFX_PIN_DC       9     // Display SPI data/command
  #define ADAGFX_PIN_RST      15    // Display Reset (255=UNUSED)
```
Note i have had this project and TFT working on Arduino Due.

The script below is the one rereferred in the first line of this post.
```c++
#include "Adafruit_GFX.h"
#include "Adafruit_ILI9341.h"
#include <Audio.h>
#include <Wire.h>
#include <SPI.h>
#include <SD.h>
#include <SerialFlash.h>

// For the ILI9341 Blue TFT.
#define TFT_CS    -1      // (no pin on TFT PCB)
#define TFT_DC    9     
#define TFT_RST   15      // (named RES on TFT PCB)

// Program Globals
Adafruit_ILI9341 tft = Adafruit_ILI9341(TFT_CS, TFT_DC, TFT_RST);

// GUItool: begin automatically generated code
AudioInputI2S            i2s1;           //xy=127,216
AudioFilterBiquad        biquad1;        //xy=286,267
AudioOutputUSB           usb2;           //xy=468,215
AudioConnection          patchCord1(i2s1, 0, usb2, 0);
AudioConnection          patchCord2(i2s1, 1, biquad1, 0);
AudioConnection          patchCord3(biquad1, 0, usb2, 1);
AudioControlSGTL5000     sgtl5000_1;     //xy=275,137
// GUItool: end automatically generated code

void setup() {
  // put your setup code here, to run once:
  delay(250);
  AudioMemory(10);
  delay(250);

  sgtl5000_1.enable();
  sgtl5000_1.inputSelect(AUDIO_INPUT_LINEIN);
  sgtl5000_1.volume(0.5);
  biquad1.setLowpass(0,500,0.707);

  SPI.setMOSI(11);
  SPI.setSCK(13);
  SPI.setMISO(12);
  tft.begin();
}

void loop() {
  // put your main code here, to run repeatedly:
  delay(20);
  
  for(uint8_t rotation=0; rotation<4; rotation++) {
    tft.setRotation(rotation);
    testText(45.5);
    delay(1000);
  }
}


void testText(double x) {
  tft.fillScreen(ILI9341_BLACK);
  tft.setCursor(0, 0);
  tft.setTextSize(4);
  tft.setTextColor(ILI9341_WHITE);
  tft.println("TEST: ");
  tft.setTextSize(7);
  tft.setTextColor(ILI9341_YELLOW); 
  tft.println(x);  
}
```
*Edit by ImpulseAdventure: added code formatting*

**Comment by Pconti31** [comment]
@asouth Simply saying your project doesn't compile isn't helpful to us or you.  Copying the actual errors messages would be helpful. 
In addition, a zip file of your project folder would allow someone to actually check your Builder setup and test a compile.

So we are left to guess what is the problem.  

One thing people have done in the past is failing to notice the Project Options tab.  You do have teensy set correctly in general settings so that when you create a new project it will default to target teensy, but if you created your project before setting this value it could be set to arduino inside your test project.   So you will need to double check your project is set for teensy.   Otherwise you will get compiler errors.

The reason for having two places is to support people that use multiple and different mcu boards and to allow them to go back to an older project and not have to reset anything in the `Edit Preferences` section like the `General` tab.

You can check this setting by opening your test project in the builder and clicking on the Project Options tab and it must show teensy for your target also.  

![profile](https://user-images.githubusercontent.com/44075106/117357216-a872a800-ae82-11eb-9bdb-1f5237e94353.png)

Of course, there could be another problem but posting the actual errors and project files will allow someone to help you out.
Paul--

**Reply by asouth** [reply]
Ah yes, i just tried it, that was it, i didn't realise the Target Platform needed to be set in PROJECT_OPTIONS if target platform differed to that specified in Edit Preferences.

I assumed the issue was related to my header files or pinout editing because there is not a Teensy 4 include file.

**Reply by ImpulseAdventure** [reply]
Thanks @asouth. Glad to hear that it is working
